### DISCUSSION TITLE
Trying to get cudu Docker Image running

### DISCUSSION THREAD

**Discussion by edeandae** [srsran/srsRAN_Project#360] [General Help] [answered]
Hi, I was trying to run the Docker Image that is published on DockerHub by softwareradiosystems on https://hub.docker.com/r/softwareradiosystems/srsran-cu-du-ubi8/tags

I downloaded the Image with the following command which should be the latest version "docker pull softwareradiosystems/srsran-cu-du-ubi8:2023.5_1.0.4"

This image was referenced in the srsRAN Project tutorial PDF page 85, but does not provide any further instructions on how to run the docker image, 

After starting the container with a simple "docker run imageID"
and the accessed it with a 
"docker exec -it <container_id> /bin/bash"

we are unsure if those previous commands need to be run with specific parameters or if we need to start services by running a "gnb -c  /opt/srsgnb/etc/gnb_rf_b210_fdd_srsUE.yml"  like command,
Note: last command errored out with a "./gnb: error while loading shared libraries: librte_kvargs.so.23: cannot open shared object file: No such file or directory"

Thanks a head of time
from
OPA5G team

**Comment by ninjab3s** [comment, marked as answer]
Hello @edeandae!

The image above is an old version of srsRAN which was uploaded for the Red Hat container registry. We have just published new images based on Ubuntu 2204 for Split 8 (with UHD) and Split 7.2 (OFH). The new docker images are available in our Docker repository [1]. We have also published the dockerfiles we use the build those images. They are located in the srsRAN Project Helm repo [2] in the `images` directory. I advice to build the images yourself instead of pulling the pre-build ones. This will ensure that the docker image you are using uses all available CPU capabilities. The following commands show how to build and run the docker image for Split 8 with UHD.

cd images/Ubuntu-2204/UHD
docker build -t srsran-split8-cudu .
docker run -it srsran-split8-cudu

Inside the docker container the srsRAN CU/DU can be run using the command `gnb -c <config file>`. Replace <config-file> with your config. The config file has to be manually created and copied into the container.

You can also use the docker-compose scripts we have added to the srsRAN Project repo [3]. You will find everything necessary in the `docker` directory.


[1] https://hub.docker.com/u/softwareradiosystems
[2] https://github.com/srsran/srsRAN_Project_helm
[3] https://github.com/srsran/srsRAN_Project
